src/app/login/page.tsx

```diff
@@ -0,0 +1,164 @@
+"use client";
+import Button from "@/components/Button";
+import Image from "next/image";
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import { useStore } from "../store";
+const toastr = require("toastr");
+
+const Login = () => {
+	/* Hooks ----------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	const router = useRouter();
+	const supabase = createClientComponentClient();
+	const { user, loginUser, signupUser } = useStore((store) => store);
+
+	const [showLogin, setShowLogin] = useState(true);
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [confirmPassword, setConfirmPassword] = useState("");
+	const [username, setUsername] = useState("");
```

Comment:
I'd love to see if we could get React Form Hook into this component somehow, would that be cool?

Also, would be keen to figure out a way to not use `useState` here on form elements if we can. As we know whenever we use a setter here it will rebuild the entire component. Open to discuss!

Reply:
Is this what you were referring to?

https://www.react-hook-form.com

Reply:
@JakeServin yeaah that's it. Also it's in the Clickup docs under Project Intro 😎

Reply:
I'd use this with Zod if you have a chance, which you can use to define a form Schema.